src/datanode/src/datanode.rs

```diff
@@ -505,6 +506,13 @@ impl DatanodeBuilder {
 
     /// Builds [KafkaLogStore].
     async fn build_kafka_log_store(config: &KafkaConfig) -> Result<Arc<KafkaLogStore>> {
+        ensure!(
+            config.max_batch_size <= ReadableSize::mb(1),
+            TooLargeMaxBatchSizeSnafu {
+                max_batch_size: config.max_batch_size
+            }
+        );
+
```

Comment:
The max limit is configurable in Kafka.

Reply:
It's not safe to remove this check. I think people prefer to use Kafka with default configurations.

Reply:
Removed this check.